src/alert-timer.jsx

```diff
@@ -14,8 +14,8 @@ export default class AlertTimer extends Component {
 		this.setupTimer(this.props.timeout, this.props.onDismiss);
 	}
 
-	componentWillReceiveProps({ timeout, onDismiss }) {
-		this.setupTimer(timeout, onDismiss);
+	componentDidUpdate() {
+		this.setupTimer(this.props.timeout, this.props.onDismiss);
```

Comment:
We only want to reset the timer if the `props.timeout` and `props.onDismiss` have changed. You should [compare to `prevProps`](https://reactjs.org/docs/react-component.html#componentdidupdate).